File: AlertDialog/app/src/main/java/com/gorkemozcan/alertdialog/MainActivity.java
package com.gorkemozcan.alertdialog;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import android.content.DialogInterface;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
//Toast message çok kısa görünüp giden silik bir yazı olacak
        //Toast.makeText(this, "Toast Message", Toast.LENGTH_LONG).show();

        //Context
        //Activity Context
        /*activity context'e erişebilmek için this kullanılır
         */
        //App Context
        /*
        app context'e erişebilmek için getApplicationContext()
        */
        //Olarak ikiye ayrılıyor
    }

    public void save(View view)
    {
        AlertDialog.Builder alert = new AlertDialog.Builder(this);

        alert.setTitle("Save");
        alert.setMessage("Are u sure?");
        alert.setPositiveButton("Yes", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {
                //save olması gerekiyor normalde ama fonksiyonu yazmadık
                Toast.makeText(getApplicationContext(), "Saved", Toast.LENGTH_LONG).show();
                    //getApplicationContext() veya MainActivity.this kullanabiliriz
                    //ancak tek başına this kulanırsak onclicklistener içersinde olduğu için hata verecektir.

            }
        });
        alert.setNegativeButton("No", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {
                Toast.makeText(MainActivity.this, "Not Saved", Toast.LENGTH_LONG).show();

            }
        });
        alert.show();
    }
}

File: KotlinAlertDialog/app/src/main/java/com/gorkemozcan/kotlinalertdialog/MainActivity.kt
package com.gorkemozcan.kotlinalertdialog

import android.content.DialogInterface
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AlertDialog

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //Context
        //Activity Context -> this
        //App Context ->applicationContext


        Toast.makeText(this@MainActivity , "Welcome", Toast.LENGTH_SHORT).show()

        /*
        button.setOnClickListener(obejct:View.OnClickListener {
            override fun onClick(v: View?) {
                Toast.makeText(this@MainActivity, "test2", Toast.LENGTH_SHORT).show()
        }
        })
         */
    }


    fun save(view: View)
    {

        val alert = AlertDialog.Builder(this)
        alert.setTitle("Save")
        alert.setMessage("Are You Sure?")
        alert.setPositiveButton("Yes"){dialog, which ->

            //Onclick Listener
            Toast.makeText(this@MainActivity, "Saved", Toast.LENGTH_SHORT).show()
        }
        alert.setNegativeButton("No"){dialog, which ->

            //Onclick Listener
            Toast.makeText(applicationContext, "Not Saved", Toast.LENGTH_SHORT).show()
        }

        alert.setNeutralButton("Cancel") {dialog, which ->
            Toast.makeText(this@MainActivity, "Cancelled", Toast.LENGTH_SHORT).show()
        }
        //Neutralbutton çok görülmese bile önemlidir. Bu uygulama özelinde yanlışlıkla buton tıklaması yapılması durumunda iptal için eklenmiştir.
        alert.show()

    }

}